mesonbuild/compilers/compilers.py

```diff
@@ -1650,9 +1689,25 @@ def openmp_flags(self):
         else:
             return ['-openmp']
 
-    def has_arguments(self, args, env, code, mode):
-        # -diag-error 10148 is required to catch invalid -W options
-        return super().has_arguments(args + ['-diag-error', '10006', '-diag-error', '10148'], env, code, mode)
+    def compiles(self, *args, **kwargs):
+        extra_args = listify(kwargs.get('extra_args') or [])
+        kwargs['extra_args'] = extra_args + [
+            '-diag-error', '10006',  # ignoring unknown option
+            '-diag-error', '10148',  # Option not supported
+            '-diag-error', '1292',   # unknown __attribute__
+        ]
+        return super().compiles(*args, **kwargs)
+
+    def get_pie_args(self):
```

Comment:
shouldn't we just add this in the `Compiler` class and shift the `self.compiler_type` into that class?

Reply:
Why would we put information specific the Intel Compiler in the generic Compiler class?